onnxruntime/core/framework/utils.cc

```diff
@@ -428,6 +428,8 @@ static void FinalizeFeedFetchCopyInfo(FeedsFetchesManager& feeds_fetches_manager
     const auto& feed = feeds[i];
     if (feed.IsTensor()) {
       feed_locations[i] = feed.Get<Tensor>().Location().device;
+    } else if (feed.IsTensorSequence()) {
+      feed_locations[i] = feed.Get<TensorSeq>().Get(0).Location().device;
```

Comment:
Here you are assuming the sequence is of Tensors... A sequence can be of sequences or tensors or maps etc... I suggest adding another check to validate this is indeed sequence of tensors

Reply:
Going and looking back at the code, I am not sure I understand this suggestion well. A `TesorSeq` is constructed by passing in several `Tensor`s and internally the `TensorSeq` stores these tensors in a vector. So, it would appear that a `TensorSeq` really is just a sequence of `Tensor`s.
Is my understanding not right?